swift/example_code/s3/basics/Tests/basics-tests/S3DemoCleanup.swift

```diff
@@ -0,0 +1,225 @@
+/*
+   A class to record the Amazon S3 buckets and files created by a code example,
+   so that they can easily be cleaned up after the example is finished.
+
+   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0
+*/
+
+import Foundation
+import AWSS3
+import ClientRuntime
+import AWSClientRuntime
+import SwiftUtilities
+
+/// Errors thrown by the `S3DemoCleanup` class.
+enum S3DemoCleanupError: Error {
+    case duplicate          /// Attempted to create an item that already exists.
+    case notFound           /// Attempted to access an item that doesn't exist.
+}
+
+/// A structure identifying a file given its Amazon S3 bucket name and the
+/// file's name within the bucket.
+public struct S3DemoFileInfo: Equatable {
+    /// The name of the bucket containing the file.
+    var bucket: String
+    /// The file's name.
+    var file: String
+    /// The contents of the file.
+    var contents: Data?
+
+    init(bucket: String, file: String, contents: Data? = nil) {
+        self.bucket = bucket
+        self.file = file
+        self.contents = contents
+    }
+
+    /// Determine if the `S3DemoFileInfo` matches the given bucket
+    /// name and file name.
+    ///1
```

Comment:
1?

Reply:
Problem with my mouse. A couple of the buttons on the little gaming controls surface on the side are too sensitive and have been randomly triggering on me, entering numbers as a result. Fixing. Thanks!